scripts/zones/Arrapago_Reef/mobs/Bukki.lua

```diff
@@ -2,10 +2,49 @@
 -- Area: Arrapago Reef
 --   NM: Bukki
 -----------------------------------
-mixins = { require("scripts/mixins/families/imp") }
+mixins =
+{
+    require("scripts/mixins/families/imp"),
+    require("scripts/mixins/job_special")
+}
+-----------------------------------
+require("scripts/globals/mobs")
 -----------------------------------
 local entity = {}
 
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:useMobAbility(1710)
+end
+
+entity.onMobFight = function(mob, target)
+    local battletime = mob:getBattleTime()
+    local jaTime = mob:getLocalVar("jaTime")
+
+    if mob:getHPP() > 20 then
+        if jaTime == 0 then
+            mob:setLocalVar("jaTime", math.random(10, 60)) -- based on captures earliest use was 10 seconds but never longer than 1 min.
+        end
+
+        if battletime >= jaTime then
+            mob:useMobAbility(1710)
+            mob:setLocalVar("jaTime", battletime + math.random(10, 60))
+        end
+    elseif mob:getHPP() < 20 then
+        if jaTime == 0 then
+            mob:setLocalVar("jaTime", math.random(5, 10)) -- based on captures earliest was one regular round then mobskill
+        end
+
+        if battletime >= jaTime then
+            mob:useMobAbility(1711)
+            mob:setLocalVar("jaTime", battletime + math.random(5, 10))
+        end
+    end
```

Comment:
I believe we can reduce the following code into:
Lemme know what you think
Also does he change the ability he uses when he's below 20% hp?
```suggestion
   local battleTime = mob:getBattleTime()

    if mob:getLocalVar("jaTime") < battleTime then
        if mob:getHPP() > 20 then
            mob:setLocalVar("jaTime", battleTime + math.random(10, 60))
            mob:useMobAbility(1710, target)
        else
            mob:setLocalVar("jaTime", battleTime + math.random(5, 10))
            mob:useMobAbility(1711, target)
        end
    end
```